drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/checks/ComparableWrapper.java

```diff
@@ -42,14 +44,8 @@ private ComparableWrapper(final Comparable value,
 
     @Override
     public int compareTo(final ComparableWrapper other) {
-        if (type.equals(Type.INFINITE)) {
-            return other.type.equals(Type.INFINITE) ? 0 : 1;
-        } else if (type.equals(Type.NEGATIVE_INFINITE)) {
-            return other.type.equals(Type.NEGATIVE_INFINITE) ? 0 : -1;
-        } else if (other.type.equals(Type.INFINITE)) {
-            return -1;
-        } else if (other.type.equals(Type.NEGATIVE_INFINITE)) {
-            return 1;
+        if(!Objects.equals(type, Type.NORMAL) || !Objects.equals(other.type, Type.NORMAL)) {
+            return type.compareTo(other.type);
```

Comment:
@jomarko So I am hard added the unit test in, but I could not get this part to work properly with INFINITE and NEGATIVE_INFINITE.

Reply:
@Rikkola sorry, do not understand, could you please clarify more what is broken thanks to my change?

Reply:
@jomarko ok, updating the original PR. Here is my change and I don't think all of these test pass with this change. https://github.com/kiegroup/drools/pull/3333/files#diff-f73f7099a394fceb38e41e0f884820a454853ff8997bfc5867e69681ff7e78d5R42-R63

Reply:
thank you @Rikkola, closing this PR as it has conflicts now, maybe I will recheck during code review